apps/shell/tash_command.c

```diff
@@ -233,28 +233,49 @@ static int tash_history(int argc, char **args)
 	return 0;
 }
 
-void tash_get_cmd_from_history(int num, char *cmd)
+/**
+ * @brief      Get the command in the history list
+ * @details    Return the address of command buffer which is selected by the index in history list
+ * @param[in]  hist_idx The index in the history list
+ * @param[out] buf      The buffer pointer to save the data selected by hist_idx
+ * @return     The address of command buffer selected on success, null on failure
+ */
+
+char *tash_get_cmd_from_history(int hist_idx, char *buf)
 {
-	if (num < 1 || num >= TASH_MAX_STORE) {
-		return;
+	int cmd_idx;
+
+	if (hist_idx < 1 || hist_idx >= TASH_MAX_STORE) {
+		printf("!%d: invalid event\n", hist_idx);
```

Comment:
hmm because it is shell command, so printf will annoy us.
Could you please change printf to some kind of verbose? it will really annoy us....=(

Reply:
This is the message Linux feedbacks to user with error case.

Reply:
You can find this message with invalid number.